src/mapping.ts

```diff
@@ -92,16 +126,19 @@ function deregisterPodLogic(id: string): void {
     oldAdmin.save();
   }
 
-  // Delete the Pod
-  store.remove('Pod', id);
-}
+  // Delete the PodUsers for this Pod
+  if (pod.users.length > 0) {
+    for (let i = 0; i < pod.users.length; i++) {
+      // Delete the PodUser
+      store.remove("PodUser", pod.users[i].toString());
+    }
+  }
 
-export function handleDeregisterPodV1_3(event: DeregisterPodV1_3): void {
-  let id = event.params.podId.toString();
-  deregisterPodLogic(id);
+  // Delete the Pod itself
+  store.remove("Pod", id);
 }
 
-export function handleDeregisterPodV1_4(event: DeregisterPodV1_4): void {
+export function handleDeregisterPod(event: DeregisterPod): void {
```

Comment:
🤔 Since this event didn't change between versions, I don't believe we need both of these, but I don't know what's best practice in this case.

Reply:
think its preferred to use consolidated event where possible